addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -276,6 +276,7 @@ export class PaymentScreen extends Component {
             // 1. Save order to server.
             syncOrderResult = await this.pos.push_single_order(this.currentOrder);
             if (!syncOrderResult) {
+                this.currentOrder.finalized = false;
```

Comment:
move this in the override of `push_single_order` after calling `pushToBlackbox`.